tests/test_colo_model_local.py

```diff
@@ -24,68 +24,135 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-
 import sys
-import warnings
 
 import pytest
 
 from smartsim import Experiment, status
+from smartsim.entity import Model
 
 if sys.platform == "darwin":
     supported_dbs = ["tcp", "deprecated"]
 else:
     supported_dbs = ["uds", "tcp", "deprecated"]
 
+def test_create_pinning_string():
+    # Automatic creation of pinned cpu list
+    assert Model._create_pinning_string(None,2) == '0,1'
+    # Individual ids only
+    assert Model._create_pinning_string([1,2],2) == '1,2'
+    # Mixed ranges and individual ids
+    assert Model._create_pinning_string([range(2),3],3) == '0,1,3'
+    # Range only
+    assert Model._create_pinning_string(range(3),3) == '0,1,2'
```

Comment:
This such a good test case to parameterize!! I demand it be parameterized!!
```suggestion
@pytest.mark.parametrize("pin_list, num_cpus, expected", [
    pytest.param(None, 2, "0,1", id="Automatic creation of pinned cpu list"),
    pytest.param([1,2], 2, "1,2", id="Individual ids only"),
    pytest.param([range(2),3], 3, "0,1,3", id="Mixed ranges and individual ids"),
    pytest.param(range(3), 3, "0,1,3", id="Range only"),
    pytest.param([range(8, 10), range(6, 1, -2)], 4, "8,9,6,4,2", id="Multiple ranges"),
])
def test_create_pinning_string(pin_list, num_cpus, expected):
    assert Model._create_pinning_string(pin_list, num_cpus) == expected
```

Reply:
LOL fair